lib/ansiblelint/errors.py

```diff
@@ -22,16 +23,33 @@ def __repr__(self):
 class MatchError(ValueError):
     """Exception that would end-up as linter rule match."""
 
-    def __init__(self, message, rule=None):
+    def __init__(
+            self,
+            message: str = "",
+            rule: Any = None,
```

Comment:
How about
```suggestion
            rule: 'AnsibleLintRule' = None,
```
and import it under `if TYPE_CHECKING`.